### ISSUE TITLE
Accesible Components

### ISSUE THREAD

**Issue by javivelasco** [react-toolbox/react-toolbox#225]
This issue will be used to track the progress on adding accessibility attributes to the components. Also, these improvements should take into consideration keyboard navigation through the items. Although it's needed for some components, there are already radio buttons, switch, autocomplete, dropdowns, etc that supports keyboard events.

@nikgraf provided [this link](http://www.w3.org/TR/wai-aria-practices) which I think is a good reference to start from.

**Comment by nathanmarks** [comment]
@javivelasco https://github.com/rackt/react-a11y

**Comment by javivelasco** [comment]
Amazing @nathanmarks, didn't know about it

**Comment by suancarloj** [comment]
Any idea when the dropdown will be accessible with keyboard?

**Comment by javivelasco** [comment]
I was planning to do work on this during next weekend. I already have a `Chip` component to decouple autocompleted fields from the autocomplete itself. Also, I'd make accesible with the keyboard in that feature. If this is going to change there no point on improving it for now!

**Comment by cometta** [comment]
```
<List shit="damn" role='presentation'  selectable ripple >
<ListSubHeader caption='Menu' />
<ListItem role='presentation' tabIndex='0'
caption='Home'
/>
</List>
```
Can help to check why` <list>` and` <ListItem>` filter out `role=`  and `tabIndex` ?

**Comment by javivelasco** [comment]
I don't understand your comment 😕

**Comment by cometta** [comment]
Sorry, let me rephrase. What I meant was, I unable to add accessibility tags to `<List> `and `<ListItem> `component. I can't even add `'id=...'`  . On the other hand, I tested with `<FontIcon> `component and there is no issue.

**Comment by javivelasco** [comment]
Yeah, this is happening because not every component is spreading every passed prop. It would be nice to have PRs when this happens because as in other many cases, this can be discovered with usage

**Comment by DevAlien** [comment]
@javivelasco any news on the dropdowns?

**Comment by javivelasco** [comment]
Not really. I'm migrating to a [different theming approach](https://github.com/react-toolbox/react-toolbox/tree/react-css-themr-migration) and I can't dedicate more time for now.

**Comment by ab712** [comment]
I set aside some time this morning to work on this today but I'm not sure how to drive out the changes from tests. It looks like the specs are creating instances of the component and only fail if an error happens on mounting the component vs doing assertions on behavior effects. Would love to start contributing to this project but I'm not sure how to test new features.

**Comment by felixzapata** [comment]
hi, maybe you can use this too: [react-axe](https://github.com/dylanb/react-axe)

**Comment by felixzapata** [comment]
@javivelasco by the way, I am working on a pull request, to fix some accessibility issues for the Dialog component.

**Comment by mhomolak** [comment]
Would it be possible to get an up to date list of all the components current made accessible? Thanks!

**Comment by javivelasco** [comment]
@felixzapata did you get any progress? @mhomolak actually I can't dedicate time but it should be easy to go through the rendered html and check if it behaves accomplishing accessibility. Anything that can be improved should be listed in a issue to be fixed. If you need to work with accesible components it would be great to get your help on this.

**Comment by felixzapata** [comment]
@javivelasco I did modifications, related with accessibility, in the Dialog component. At this moment I have all the changes in [my fork](https://github.com/felixzapata/react-toolbox/commits/a11y-dialog). It is not fully updated with yours. 

The problems I have is my lack of knowledge of React in order to make some things. Now the Dialog is more accessible, I have in my todo list to check the Date component, the Calendar (I think) component, ... I mean, all the components who depend from the Dialog. 

Because I can´t (I think) upload the Dialog without update the other components due to the changes I did on it.

How many elements use the Dialog component?

**Comment by javivelasco** [comment]
I think only pickers. Anyway, any change on accessibility should be done over #666. We can start by supporting `aria` attributes. Maybe it's a good idea to open a new issue listing checkboxes to track the progress.

**Comment by felixzapata** [comment]
I know. I have to update my fork sooner or later but for now (and because I know my changes) I did not do that yet.

I´ve started with the Dialog because I though it was a critical component and relatively easy to fix. But for example, for this component is not only adding `aria-attributes`. It is necessary to manage the focusable elements when the modal dialog is opened.

**Comment by jdbence** [comment]
I've been working on making the tabs accessible in [this branch](https://github.com/jdbence/react-toolbox/tree/feature/aria). Any feedback would be great.
- [x] Works with multiple Tabs components on one page
- [x] Navigate tabs with keyboard arrows that bound wrap
- [x] Adds correct aria attributes for voice
- [ ] Take into account disabled tabs
- [ ] Allow aria descriptions

**Comment by felixzapata** [comment]
@javivelasco the PR #666 has not been merged yet? is it correct?

**Comment by felixzapata** [comment]
I will make modifications for the `iconMenu` too.

**Comment by felixzapata** [comment]
[I´ve started a branch](https://github.com/felixzapata/react-toolbox/commits/a11y-iconmenu) about the `iconMenu` to fix issues related with the accessibility.

**Comment by felixzapata** [comment]
Hi, I´ve created a pull request for the `iconMenu` component.

**Comment by felixzapata** [comment]
Hi, I´ve created new pull requests: #1100, #1101 and #1102

**Comment by felixzapata** [comment]
Hi @jdbence, I've not seen that you have been working on `FontIcon` too, sorry. I am going to work on the `Snackbar` component.

**Comment by kozakluke** [comment]
any progress with navigate Dropdown with the keyboard?

**Comment by javivelasco** [comment]
I have something done in a different project. The reimplementation of Dropdown and Autocomplete will head it

**Comment by tconroy** [comment]
hey @javivelasco - sorry to bother, just curious if any progress has been made for dropdown keyboard navigation? Thanks for all your hard work!

**Comment by javivelasco** [comment]
I'm just tweeted this: https://twitter.com/javivelasco/status/912372144285286401 

It will be used in the next version for Dropdowns, Autocompletes, Tables, Lists and any other "listable" thing :D
